src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,2011 @@
+
+        /// <inheritdoc cref="INumberBase{TSelf}.MaxMagnitudeNumber(TSelf, TSelf)" />
+        public static BFloat16 MaxMagnitudeNumber(BFloat16 x, BFloat16 y)
+        {
+            // This matches the IEEE 754:2019 `maximumMagnitudeNumber` function
+            //
+            // It does not propagate NaN inputs back to the caller and
+            // otherwise returns the input with a larger magnitude.
+            // It treats +0 as larger than -0 as per the specification.
+
+            BFloat16 ax = Abs(x);
+            BFloat16 ay = Abs(y);
+
+            if ((ax > ay) || IsNaN(ay))
+            {
+                return x;
+            }
+
+            if (ax == ay)
+            {
+                return IsNegative(x) ? y : x;
+            }
+
+            return y;
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.MinMagnitude(TSelf, TSelf)" />
+        public static BFloat16 MinMagnitude(BFloat16 x, BFloat16 y) => (BFloat16)MathF.MinMagnitude((float)x, (float)y);
+
+        /// <inheritdoc cref="INumberBase{TSelf}.MinMagnitudeNumber(TSelf, TSelf)" />
+        public static BFloat16 MinMagnitudeNumber(BFloat16 x, BFloat16 y)
+        {
+            // This matches the IEEE 754:2019 `minimumMagnitudeNumber` function
+            //
+            // It does not propagate NaN inputs back to the caller and
+            // otherwise returns the input with a larger magnitude.
+            // It treats +0 as larger than -0 as per the specification.
+
+            BFloat16 ax = Abs(x);
+            BFloat16 ay = Abs(y);
+
+            if ((ax < ay) || IsNaN(ay))
+            {
+                return x;
+            }
+
+            if (ax == ay)
+            {
+                return IsNegative(x) ? x : y;
+            }
+
+            return y;
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.TryConvertFromChecked{TOther}(TOther, out TSelf)" />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static bool INumberBase<BFloat16>.TryConvertFromChecked<TOther>(TOther value, out BFloat16 result)
+        {
+            return TryConvertFrom(value, out result);
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.TryConvertFromSaturating{TOther}(TOther, out TSelf)" />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static bool INumberBase<BFloat16>.TryConvertFromSaturating<TOther>(TOther value, out BFloat16 result)
+        {
+            return TryConvertFrom(value, out result);
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.TryConvertFromTruncating{TOther}(TOther, out TSelf)" />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static bool INumberBase<BFloat16>.TryConvertFromTruncating<TOther>(TOther value, out BFloat16 result)
+        {
+            return TryConvertFrom(value, out result);
+        }
+
+        private static bool TryConvertFrom<TOther>(TOther value, out BFloat16 result)
```

Comment:
The `TryConvertTo` paths have a callout about non-first class type and ensuring that all built-ins are handled. This does not and is missing some types?

Reply:
This still looks like its missing some types that `float`, `double`, and `Half` all handle.

Believe the handling was also missing for `System.Runtime.CompilerServices.NFloat`

Reply:
I've checked that other native float types aren't currently handling nfloat. Only nfloat is handling other types in a single direction.

Reply:
> I've checked that other native float types aren't currently handling nfloat.

Yes, which means `nfloat` needs to be updated to add the `BFloat16` paths.

> Only nfloat is handling other types in a single direction.

`nfloat` should be handling types bi-directionally, much as `Complex` and `BigInteger` do, since they exist in a different layer.

Reply:
I should already have `nfloat`, `Complex` and `BigInteger` updated.